File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals)<=1:
            return intervals
        intervals.sort(key=lambda x:x[0])
        result=[intervals[0]]
        for cur in  intervals[1:]:
            last=result[-1]
            if cur[0]<=last[1]:
                last[1]=max(last[1],cur[1])
            else:
                result.append(cur)
        return result

File: 56-merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals,(a,b)->Integer.compare(a[0], b[0]));
        List<int[]> merged=new ArrayList<>();
        int[] cur=intervals[0];
        merged.add(cur);
        for(int[] last:intervals){
        if(cur[1]>=last[0]){
            cur[1]=Math.max(cur[1],last[1]);
        }
        else{
            cur=last;
            merged.add(cur);
        }
        }
         return merged.toArray(new int[merged.size()][]);

    }
}